packages/backup/service.go

```diff
@@ -34,6 +34,14 @@ func NewBackupService(backupClient backupInterface, restoreClient restoreInterfa
 	}
 }
 
+func (s *DefaultBackupService) DeleteBackup(ctx context.Context, req *pbBackup.DeleteBackupRequest) (*pbBackup.DeleteBackupResponse, error) {
+	err := s.backupClient.Delete(ctx, req.Name, metav1.DeleteOptions{})
+	if err != nil {
+		return nil, err
```

Comment:
```suggestion
		return nil, fmt.Errorf("failed to delete backup: %w", err)
```

add some context to the error